### ISSUE TITLE
[ENHANCEMENT] Show printer name in Prusa Link

### ISSUE THREAD

**Issue by neuron-whisperer** [prusa3d/Prusa-Firmware-Buddy#4302]
### Printer model

All

### Describe the enhancement

This one is really, really simple.

I own two printers (a MK4S recently upgraded from a MK3S, and a MK4 with an upgrade kit incoming). When I access Prusa Link via Wi-Fi to feed them models, the page shows me a lot of stuff about the printer... *except its name.* I have to check out the IP address to make sure I'm talking to the right printer.

Can you please just put the name of the printer somewhere on the page? Thanks.

**Comment by danopernis** [comment]
By name I assume you mean printer model? There is also a hostname, which the browser will happily show in the address bar :wink: Other than that I don't know about any other _name_ you are referring to.

This can be implemented but our schedule is really tight right now. We are willing to accept patches though. Note that this probably needs to be implemented also in the https://github.com/prusa3d/Prusa-Link-Web repository.

**Comment by neuron-whisperer** [comment]
Oh! Great - I'll work on it.

Printers have user-assigned names stored in firmware. Mine are named Protoceratops and Protohadros, and when PrusaSlicer offers to send a model directly to a printer, it shows me a status page *including names,* like this:

![Screenshot 2024-11-20 at 8 10 44 AM](https://github.com/user-attachments/assets/339417b5-d708-4a70-a9bc-64dee9ee0423)

I'm merely asking for the same information to be displayed in the PrusaLink page. Should be very easy as the printer itself is serving the page!

I understand that you can't get to it; I'll be happy to roll up my sleeves and see if I can implement it.

**Comment by bkerler** [comment]
This is already integrated into Prusa-Link-Web, it just hasn't made it into the firmware yet.

**Comment by neuron-whisperer** [comment]
I'm still interested in this and bumping it to keep it in the queue.

**Comment by bkerler** [comment]
@neuron-whisperer PrusaLink is already showing the printer name (hostname) :

![Image](https://github.com/user-attachments/assets/9166ac85-ea47-4724-b6eb-fde656500eca)

Or do you mean something different ?

**Comment by neuron-whisperer** [comment]
My dashboard has no hostname field.

Here are the dashboard pages for each of my printers:

![Image](https://github.com/user-attachments/assets/da5554fb-c91d-4a83-8bbf-d9c1743f5624)

![Image](https://github.com/user-attachments/assets/e5cc409f-a88f-4300-a2d4-788db6494de5)

Literally the only way to tell which printer I'm using is to look at the address bar. Fortunately, I'm using DHCP and I know which IP address maps to which printer, but that's a terrible solution to this problem.

I can't find any way to customize the Prusa Link page for either printer to show different information.

**Comment by bkerler** [comment]
Ok, interesting. For me, I see the information (using the latest 6.2.4 rc1). Are you using the latest fw ?

**Comment by neuron-whisperer** [comment]
Both printers are on 6.2.3+8862. I don't run release candidates but full releases. Maybe it's being added in 6.2.4.

**Comment by bkerler** [comment]
Yeah, could be. Let's wait for the release, then we can figure out if the name is displayed.

**Comment by neuron-whisperer** [comment]
Nope. This is still happening in 6.2.4.

And I would be happy to upload photos showing (1) that the printer is on 6.2.4 and (2) the PrusaLink page still doesn't show the name - but for some reason, GitHub now won't allow uploads to this post.

**Comment by 3d-gussner** [comment]
FYI: It should be fixed in next major release.

**Comment by paulcobbaut** [comment]
Same issue here. I now have two MK4 printers on DHCP and no way to see the hostname in Prusalink. Firmware is up to date.

**Comment by Prusa-Support** [comment]
The fix seems to be currently active in the last firmware version.
Please give it a try. Final feedback would be appreciated before closing the issue.

Michele Moramarco
Prusa Research

**Comment by paulcobbaut** [comment]
Can confirm that it now works! Thanks!!
